Tests/DatadogTests/Datadog/Tracing/Utils/MethodSwizzlerTests.swift

```diff
@@ -0,0 +1,163 @@
+        let newImp = imp_implementationWithBlock(newBlockImp)
+
+        let obj = BaseClass()
+
+        XCTAssertNoThrow(try swizzler.swizzle(selector: sel, in: BaseClass.self, with: newImp))
+        XCTAssertEqual(obj.perform(sel)?.takeUnretainedValue() as? String, swizzledReturnValue)
+    }
+
+    func testUnswizzling() {
+        typealias TypedIMP = @convention(c) (AnyObject, Selector) -> String
+        typealias TypedBlockIMP = @convention(block) (AnyObject) -> String
+
+        let sel = #selector(BaseClass.methodToSwizzle as (BaseClass) -> () -> String)
+        let swizzledReturnValue = "this is swizzled imp"
+        let newBlockImp: TypedBlockIMP = { impSelf -> String in
+            return swizzledReturnValue
+        }
+        let newImp = imp_implementationWithBlock(newBlockImp)
+
+        try! swizzler.swizzle(selector: sel, in: BaseClass.self, with: newImp)
+
+        let obj = BaseClass()
+
+        XCTAssertNoThrow(try swizzler.unswizzle(selector: sel, in: BaseClass.self))
+        XCTAssertEqual(obj.perform(sel)?.takeUnretainedValue() as? String, BaseClass.returnValue)
+    }
+
+    func testSwizzleSuperclassMethod() {
+        typealias TypedIMP = @convention(c) (AnyObject, Selector) -> String
+        typealias TypedBlockIMP = @convention(block) (AnyObject) -> String
+
+        let klass: AnyClass = EmptySubclass.self
+        let sel = #selector(EmptySubclass.methodToSwizzle as (EmptySubclass) -> () -> String)
+        let swizzledReturnValue = "this is swizzled imp"
+        let newBlockImp: TypedBlockIMP = { impSelf -> String in
+            return swizzledReturnValue
+        }
+        let newImp = imp_implementationWithBlock(newBlockImp)
+
+        do {
+            try swizzler.swizzle(selector: sel, in: klass, with: newImp)
+        } catch SwizzlingError.methodNotFound(let unfoundSelector, let searchedClassName) {
+            XCTAssertEqual(unfoundSelector, NSStringFromSelector(sel))
+            XCTAssertEqual(searchedClassName, NSStringFromClass(klass))
+        } catch {
+            XCTAssertNil(error)
+        }
+    }
+
+    func testSwizzleNonNSObjectSubclass() {
+        typealias TypedIMP = @convention(c) (AnyObject, Selector) -> String
+        typealias TypedBlockIMP = @convention(block) (AnyObject) -> String
+
+        let klass: AnyClass = NonNSObjectSubclass.self
+        let sel = #selector(NonNSObjectSubclass.methodToSwizzle as (NonNSObjectSubclass) -> () -> String)
+        let swizzledReturnValue = "this is swizzled imp"
+        let newBlockImp: TypedBlockIMP = { impSelf -> String in
+            return swizzledReturnValue
+        }
+        let newImp = imp_implementationWithBlock(newBlockImp)
+
+        do {
+            try swizzler.swizzle(selector: sel, in: klass, with: newImp)
+        } catch SwizzlingError.classIsNotNSObjectSubclass(let className) {
+            XCTAssertEqual(className, NSStringFromClass(klass))
+        } catch {
+            XCTAssertNil(error)
+        }
```

Comment:
Same thing as for `testSwizzleSuperclassMethod()` - let's increase this test's readability.